src/Editor/Editor.cpp

```diff
@@ -365,6 +371,79 @@ void Editor::Show(float deltaTime) {
     if (ImGui::GetMouseCursor() < 5) {
         glfwSetCursor(MainWindow::GetInstance()->GetGLFWWindow(), cursors[ImGui::GetMouseCursor()]);
     }
+
+    // Widget Controller for translation, rotation  and scale.
+    ImGuizmo::BeginFrame();
+    ImGuizmo::Enable(true);
+    //Widget operation is in local mode
+    ImGuizmo::MODE currentGizmoMode(ImGuizmo::LOCAL);
+    ImGuiIO& io = ImGui::GetIO();
+
+    // Check that there is an active Entity.
+
+    // Get current active Entity.
+    glm::mat4 currentEntityMatrix = glm::mat4();
+
+    Entity* currentEntity = resourceView.GetScene().entityEditor.GetEntity();
+
+    if (resourceView.GetScene().entityEditor.GetEntity() != NULL) {
```

Comment:
We should perform the comparison on currentEntity instead of getting the same thing again to prevent possible future bugs.